### DISCUSSION TITLE
🤔: 「プッシュ通知をしたいかどうか」は、ネイティブアプリ採用の根拠になりますか？

### DISCUSSION THREAD

**Discussion by ghost** [ws-4020/mobile-app-crib-notes#112] [Q&A] [answered]
コンテンツの対象範囲から外れる質問になってしまうのですが、携帯の通知領域に何かを表示したいというケースで、webプッシュでも実現できそうと思います。

コンテンツで紹介されているプッシュ通知とwebプッシュのどちらを使うべきか、悩んでいます。
ウェブアプリとネイティブアプリのどちらで開発するかを決めるときに、「プッシュ通知をしたいかどうか」は判断材料や選択肢を狭める要因にあまりならないと思いましたが、的外れでしょうか？

---

回答に対するコメント

webのほうがリリースが楽なので魅力的かなと思ったのですが、safariがNGなのは残念です。

---

※ コンテンツを確認してもらった後の質問とやりとりを転記します

**Comment by ghost** [comment, marked as answer]
ネイティブで作るかどうかの判断基準としてプッシュ通知だけが決め手になることはないと思います。

ただ、[Push API - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Push_API)や[Can I useのPush API](https://caniuse.com/push-api)を確認すると、iOS（Safari on iOS）は2021-04-05時点ではWebプッシュをサポートしていないようです。なので、iOSへのプッシュ通知が必須要件となると（PWAではなく）ネイティブアプリで開発する必要があると思います。

**Reply by ghost** [reply]
（質問者からのコメントを転記）
webのほうがリリースが楽なので魅力的かなと思ったのですが、safariがNGなのは残念です。

**Reply by ghost** [reply]
たしかにリリースが楽なことは魅力的なので、iOSへのプッシュ通知とのトレードオフをどうするか、検討の余地はあると思います。

**Comment by creature-water-valley** [comment]
回答が付いているのでクローズ
